barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/goblin_ultra_flavor.hpp

```diff
@@ -260,28 +264,72 @@ class GoblinUltraFlavor {
         auto get_shifted() { return ShiftedEntities<DataType>::get_all(); };
     };
 
+    /**
```

Comment:
Unfortunately all of the flavor files have this misleadingly large diff that's due that's largely due to the need to move the definition of ProverPolynomials (and AllValues) prior to the definition of ProvingKey (since ProvingKey now has a ProverPolynomials). There are some differences in the ProverPolynomials classes though, namely the constructor that inits all polys to the zero poly (an analog to what used to be done in the base class `ProvingKey_`) and the method `set_shifts()`

Reply:
You didn't give a diff on the benchmark--is there really only negligible cost (possibly bc we were already initing to 0) to this change?